vermouth/tests/test_links.py

```diff
@@ -23,156 +23,157 @@
-        (('>>', '<<<'), (5, 5), False),
-        (('<<', '>>>'), (3, 5), True),
-        (('<<', '>>>'), (3, 3), False),
-        (('<<', '>>>'), (3, 1), False),
+        (('>', '>'), (3, 3), ('A', 'A'), True),
+        (('>', '>'), (3, 4), ('A', 'A'), False),
+        (('>', '>'), (3, 1), ('A', 'A'), False),
+        (('>>>', '>>>'), (3, 3), ('A', 'A'), True),
+        (('>>>', '>>>'), (3, 4), ('A', 'A'), False),
+        (('>>>', '>>>'), (3, 1), ('A', 'A'), False),
+        (('<', '<'), (3, 3), ('A', 'A'), True),
+        (('<', '<'), (3, 4), ('A', 'A'), False),
+        (('<', '<'), (3, 1), ('A', 'A'), False),
+        (('<<<', '<<<'), (3, 3), ('A', 'A'), True),
+        (('<<<', '<<<'), (3, 4), ('A', 'A'), False),
+        (('<<<', '<<<'), (3, 1), ('A', 'A'), False),
+        (('>', '>>'), (3, 6), ('A', 'A'), True),
+        (('>', '>>'), (3, 3), ('A', 'A'), False),
+        (('>', '>>'), (3, 1), ('A', 'A'), False),
+        (('>>', '>>>'), (3, 6), ('A', 'A'), True),
+        (('>>', '>>>'), (3, 3), ('A', 'A'), False),
+        (('>>', '>>>'), (3, 1), ('A', 'A'), False),
+        (('<', '<<'), (5, 3), ('A', 'A'), True),
+        (('<', '<<'), (5, 5), ('A', 'A'), False),
+        (('<', '<<'), (5, 7), ('A', 'A'), False),
+        (('>>', '<<<'), (5, 3), ('A', 'A'), True),
+        (('>>', '<<<'), (5, 5), ('A', 'A'), False),
+        (('<<', '>>>'), (3, 5), ('A', 'A'), True),
+        (('<<', '>>>'), (3, 3), ('A', 'A'), False),
+        (('<<', '>>>'), (3, 1), ('A', 'A'), False),
         # ><, *  no comparison
-        (('>', '*'), (3, 3), True),
-        (('>>', '***'), (3, 2), True),
-        (('<<<', '**'), (2, 5), True),
+        (('>', '*'), (3, 3), ('A', 'A'), True),
+        (('>>', '***'), (3, 2), ('A', 'A'), True),
+        (('<<<', '**'), (2, 5), ('A', 'A'), True),
         # *, 0
-        (('***', 0), (4, 5), True),
-        (('**', 0), (5, 4), True),
-        (('****', 0), (5, 5), False),
+        (('***', 0), (4, 5), ('A', 'A'), True),
+        (('**', 0), (5, 4), ('A', 'A'), True),
+        (('****', 0), (5, 5), ('A', 'A'), False),
         # *, n  no comparison
-        (('***', 1), (4, 5), True),
-        (('**', 2), (5, 4), True),
-        (('****', -3), (5, 5), True),
+        (('***', 1), (4, 5), ('A', 'A'), True),
+        (('**', 2), (5, 4), ('A', 'A'), True),
+        (('****', -3), (5, 5), ('A', 'A'), True),
         # *, ><  no comparison
-        (('***', '<<'), (4, 5), True),
-        (('**', '>'), (5, 4), True),
-        (('****', '>>>'), (5, 5), True),
+        (('***', '<<'), (4, 5), ('A', 'A'), True),
+        (('**', '>'), (5, 4), ('A', 'A'), True),
+        (('****', '>>>'), (5, 5), ('A', 'A'), True),
         # *, *
-        (('*', '*'), (6, 6), True),
-        (('*', '*'), (6, 5), False),
-        (('**', '**'), (6, 6), True),
-        (('**', '**'), (6, 5), False),
-        (('**', '****'), (5, 7), True),
-        (('**', '****'), (5, 5), False),
+        (('*', '*'), (6, 6), ('A', 'A'), True),
```

Comment:
Could you add a `(('*', '*'), (6, 6), ('A', 'B'), True),`? Or `((0, '*'), ...)`? One of those should match but didn't before this PR.